packages/contracts/test/framework/plugin/plugin-setup-processor.ts

```diff
@@ -260,98 +322,151 @@ describe.skip('Plugin Setup Processor', function () {
+        const pluginRepoPointer: PluginRepoPointer = [repoU.address, 1, 4];
+
+        const {preparedSetupId} = await prepareInstallation(
           psp,
           targetDao.address,
-          pluginSetupBad,
-          repoU.address,
-          data1
-        );
-
-        const data2 = ethers.utils.defaultAbiCoder.encode(
-          ['address'],
-          [plugin]
+          pluginRepoPointer,
+          '0x'
         );
 
         await expect(
           psp.prepareInstallation(
             targetDao.address,
-            pluginSetupBad,
-            repoU.address,
-            data2
+            createPrepareInstallationParams(pluginRepoPointer, '0x')
           )
-        ).to.be.revertedWithCustomError(psp, 'SetupAlreadyPrepared');
+        )
+          .to.be.revertedWithCustomError(psp, 'SetupAlreadyPrepared')
+          .withArgs(preparedSetupId);
       });
 
-      it('prepares an UUPS upgradeable plugin installation', async () => {
-        let plugin;
-        let helpersUV1;
-        let permissionsUV1;
-        ({
-          plugin: plugin,
-          helpers: helpersUV1,
-          permissions: permissionsUV1,
-        } = await prepareInstallation(
+      it('reverts if plugin with the same address is already installed.', async () => {
+        // uses plugin setup that returns the same plugin address and dependencies
+        // each time you call it. Useful to generate the same plugin address
+        // which should revert.
+        const pluginRepoPointer: PluginRepoPointer = [repoU.address, 1, 4];
+
+        await installPlugin(
           psp,
           targetDao.address,
-          setupUV1.address,
-          repoU.address,
+          pluginRepoPointer,
           EMPTY_DATA
-        ));
+        );
 
-        expect(plugin).to.not.equal(AddressZero);
-        expect(helpersUV1).to.deep.equal(mockHelpers(1));
-        expect(permissionsUV1).to.deep.equal(
-          mockPermissionsOperations(0, 1, Operation.Grant).map(perm =>
-            Object.values(perm)
+        await expect(
+          psp.prepareInstallation(
+            targetDao.address,
+            createPrepareInstallationParams(pluginRepoPointer, '0x')
           )
-        );
+        ).to.be.revertedWithCustomError(psp, 'PluginAlreadyInstalled');
       });
 
-      it('prepares a cloneable plugin installation', async () => {
-        let plugin;
-        let helpersCV1;
-        let permissionsCV1;
-        ({
-          plugin: plugin,
-          helpers: helpersCV1,
-          permissions: permissionsCV1,
-        } = await prepareInstallation(
+      // 1. prepareInstall for pluginId1 => setupId1
+      // 2. applyInstall for pluginId1 => setupId1
+      // 3. uninstall the plugin with applyUninstall.
+      // 4. prepareInstall for pluginId1 => setupId1 which succeeds.
+      it('EDGE-CASE: allows to prepare plugin installation with the same address and setupId if it was installed and then uninstalled', async () => {
```

Comment:
Isn't this a "Reinstallation"? Think if the test naming can be improved
```suggestion
      it('reinstalls a plugin contract (same address and `prepareSetupId` that has been uinstalled previously', async () => {
```